package.json

```diff
@@ -54,11 +53,14 @@
     "@chiragrupani/karma-chromium-edge-launcher": "2.1.0",
     "@jest/core": "26.6.3",
     "@jridgewell/doctrine": "3.0.2",
+    "@percy/cli": "1.0.0-beta.39",
+    "@percy/puppeteer": "2.0.0",
```

Comment:
These two deps and the one below are large and should continue to live in the `visual-diff` directory. We shouldn't force every `amphtml` developer to install them. Curious about why they had to be moved out.

Reply:
Saw your other comment. I still think visual-diff specific packages should not be moved to the root package. This is the model followed by every large testing tool / task.

Reply:
See the comment below that took me to long to write 😅

Reply:
For context:
```log
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ rm -rf node_modules/
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ npm ci

> amp-html@0.1.0 preinstall
> node build-system/common/check-package-manager.js

Detected node version v14.16.0 (latest LTS).
Detected gulp version 4.0.2.
Detected python version 3.8.7.
Detected npm version 7.6.1. Installing packages...

added 3647 packages in 17s
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ du -h --summarize node_modules/puppeteer/
4.4M	node_modules/puppeteer/
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ du -h --summarize node_modules/@percy/
3.8M	node_modules/@percy/
```

Reply:
I strongly disagree, that model introduced an unnecessary level of complexity to our code - individual `gulp` tasks should not have to handle installing npm package...

Reply:
I like simpler code!! But I gotta agree with Raghu because the npm packages from all tasks will add up. Just off the top of my head, e2e has selenium-webdriver, chromedriver, geckodriver.

To be fair we should have probably added comments to the tasks to clarify.

Reply:
I see what you're getting at. Let me offer the full rationale:

- `amphtml` is a monorepo with a [dozen](https://github.com/ampproject/amphtml/search?q=%22package%22+in%3Apath&type=code) `package.json` files across sub-packages and sub-tasks.
- For deps that are part of a majority of developers' workflows, we've generally stuck them in the root `package.json`.
- While this has worked in the past, the directory size of `node_modules` is approaching 1 GB. There are two ways of dealing with this (that are actively being pursued):
    - Completely remove or replace outdated or unnecessary dependencies.
    - Move task-specific dependencies that aren't part of the default dev workflow out of the monorepo root.
- Every extra minute of work done during a CI job (e.g. installing packages) translates to half a million minutes of work (with a non-trivial $$ cost) across all CI jobs over a year. Moving deps needed by just one task / CI job will transfer the cost to all the other ~20 parallel jobs that don't need it.
- In the case of `tasks/visual-diff`, percy / puppeteer are task specific, and the entry in the root `.npmrc` is a detail of implementation that the root package installer should not have to worry about.
- Finally, for the correctness of CI checks, I think we should still have `puppeteer` download a known version of Chrome determined by its own version (or by a known override). This will help avoid non-deterministic results in future, say when a new version of Chrome with breaking changes is released.

**TL;DR:** My comment ended up longer and my push back stronger than I originally intended (sorry!), but I think there are enough good reasons to keep the current dependency model 😃

Reply:
Alright, reverted that part of the change